sim/dex-data.ts

```diff
@@ -102,14 +116,12 @@ export class BasicEffect implements EffectData {
 	sourceEffect: string;
 
 	constructor(data: AnyObject) {
-		this.exists = true;
-		Object.assign(this, data);
-
 		this.name = Utils.getString(data.name).trim();
 		this.id = data.realMove ? toID(data.realMove) : toID(this.name); // Hidden Power hack
 		this.fullname = Utils.getString(data.fullname) || this.name;
 		this.effectType = Utils.getString(data.effectType) as EffectType || 'Condition';
-		this.exists = !!(this.exists && this.id);
+		const exists = Object.prototype.propertyIsEnumerable.call(data, 'exists') ? data.exists : true;
+		this.exists = !!(exists && this.id);
```

Comment:
Fair enough, no, I don't need those semantics.

```suggestion
		this.exists = !!(data.exists !== true && this.id);
```

Reply:
Or, actually, nullish coalescing exists now, this is probably way more readable.

```suggestion
		const exists = Object.prototype.propertyIsEnumerable.call(data, 'exists') ? data.exists : true;
		this.exists = data.exists ?? !!this.id;
```